main/src/main/scala/sbt/Cross.scala

```diff
@@ -261,6 +312,34 @@ object Cross {
 
     type ScalaVersion = String
 
+    def resolveNightlyCB(major: String): String = {
+      val url =
+        s"https://raw.githubusercontent.com/scala/community-builds/$major.x/nightly.properties"
+      val props = new java.util.Properties
+      props.load(new java.net.URL(url).openStream)
+      props.getProperty("nightly").ensuring(_ != null)
+    }
+
+    def resolveNightly(major: String): String = {
+      val url =
+        s"https://api.travis-ci.org/repo/scala%2Fscala/builds?limit=1&state=passed&include=build.commit&branch.name=$major.x"
+      val connection = new java.net.URL(url).openConnection()
+      connection.addRequestProperty("Travis-API-Version", "3")
+      val source = scala.io.Source.fromInputStream(connection.getInputStream())(scala.io.Codec.UTF8)
+      val content = source.getLines().mkString
+      val pattern = """"sha":\s+"([a-z0-9]{40})"""" r
+      val u = pattern.unanchored
+      val minor = resolveNightlyCB(major).takeWhile(_ != '-').split('.')(2)
```

Comment:
This is pretty gnarly, even compared to the rest of it.

Getting releases from the github API may be better.